vmware-base/src/main/java/com/cloud/hypervisor/vmware/mo/HostMO.java

```diff
@@ -1156,6 +1194,41 @@ public String getNetworkName(String netMorVal) throws Exception {
         return networkName;
     }
 
+    public void createPortGroup(HostPortGroupSpec spec) throws Exception {
+        if (spec == null) {
+            return;
+        }
+
+        synchronized (_mor.getValue().intern()) {
```

Comment:
can you explain why you synchronise on this object @sureshanaparti ? Intuitively I'd say just on the `HostMO` instance you want to add to the portgoup, i.e. `_mor`

Reply:
@DaanHoogland `_mor` object val might not be the same / unique while attempting this from multiple threads. intern() call on its value ensures same val of the String object, and so valid object for synchronisation.

Reply:
tnx @sureshanaparti :+1: